process_incoming_data.py

```diff
@@ -0,0 +1,604 @@
+              36: "NY",
+              37: "NC",
+              38: "ND",
+              39: "OH",
+              40: "OK",
+              41: "OR",
+              42: "PA",
+              44: "RI",
+              45: "SC",
+              46: "SD",
+              47: "TN",
+              48: "TX",
+              49: "UT",
+              50: "VT",
+              51: "VA",
+              53: "WA",
+              54: "WV",
+              55: "WI",
+              56: "WY",
+              }
+
+## Methods
+
+def save_csv(filename, content, writemode='wb'):
+    """Saves the specified content object into a csv file."""
+    if not os.path.exists(os.path.dirname(filename)):
+        os.makedirs(os.path.dirname(filename))
+        print("Created directories for {}".format(os.path.dirname(filename)))
+
+    with open(filename, writemode) as csvfile:
+        writer = csv.writer(csvfile, delimiter=',')
+        writer.writerows(content)
+
+    return True
+
+
+def load_csv(filename, skipheaderrow=True):
+    """Loads CSV data from a file"""
+    with open(filename, 'rb') as csvfile:
+        reader = csv.reader(csvfile)
+        # TODO: Handle header row, consider creating list of dictionaries with DictReader
+        data = list(reader)
+
+    if skipheaderrow:
+        return data[1:]
+    else:
+        return data
+
+
+def load_paths(inputpath=DEFAULT_INPUT_FOLDER, outputpath=DEFAULT_OUTPUT_FOLDER):
+    """Loads the root path and destination paths and performs path checking"""
+    inpath = expand_path(inputpath)
+    outpath = expand_path(outputpath)
+    # TODO: Make sure paths exist
+
+    # TODO: Do we want the path made or throw an error if it doesn't exist?
+    # try:
+    #     os.makedirs(destpath)
+    # except:
+    #     # Exception occurs when attempting to makedirs that exist
+    #     # Ignore this exception
+    #     # TODO: Parse for specific exception type so we don't ignore
+    #     # unexpected exceptions
+    #     pass
+
+    return inpath, outpath
+
+
+def expand_path(path):
+    rootpath = os.path.abspath(os.path.expanduser(path))
+
+    return rootpath
+
+
+def get_csv_list(path):
+    """Loads a list of files in the specified directory"""
+    files = [f for f in os.listdir(path)
+             if f.lower().endswith('.csv') 
+             and os.path.isfile(os.path.join(path, f))]
+
+    return files
+
+
+## Program flow
+
+def process_data_files(inputpath, outputpath, report_success=True, report_failure=True):
+    """Processes raw data from the Office of Research"""
+    # Get a list of files in the raw data directory
+    inputfiles = get_csv_list(inputpath)
+    successes = []
+    failures  = []
+
+    # For each file, open and munge data
+    for filename in inputfiles:
+        # Check for market in filename
+        market = find_market(filename)
+        if market is None:
+            print("Found file '{}' does not specify market".format(filename))
+            failures.append(filename)
+            continue
```

Comment:
Is it important to know the failures in real-time? If not, it might be nice to record the file and the reason for failure as tuples in the `failures` list so that you can print them in a single place (L233) below. But that would just be my preference. It's fine like it is too.